### ISSUE TITLE
Tensor slice

### ISSUE THREAD

**Issue by zongzi3zz** [Project-MONAI/MONAI#5353]
**Describe the bug**
x[mask >= threshold] would raise: _TypeError: only integer tensors of a single element can be converted to an index_
type(x):MetaTensor, type(mask):MetaTensor
x: [B, N, C], mask: [B, N]

**Comment by KumoLiu** [comment]
Hi @zongzi3zz, it seems it is caused by your Inconsistent shape.
You can try it like this, it works when data and mask are all MetaTensor.
```
import torch
from monai.data import MetaTensor

data = MetaTensor(torch.zeros([64,64,3]))
mask = MetaTensor(torch.zeros([64,64,3]))
mask[:, 0, :] = 2

data[mask >1 ] =1
print(type(data), type(mask))

output --> <class 'monai.data.meta_tensor.MetaTensor'> <class 'monai.data.meta_tensor.MetaTensor'>
```
Hope it can help you, thanks!

**Comment by zongzi3zz** [comment]
Thank you! I try it again, it is not caused by your Inconsistent shape. And I find the case only happens when
MetaTensor have extra info like


if the MetaTensor only has


The Error wouldn't happen.
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Project-MONAI/MONAI"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年10月19日(星期三) 上午10:25
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Project-MONAI/MONAI] Tensor slice (Issue #5353)





 
Hi @zongzi3zz, it seems it is caused by your Inconsistent shape.
 You can try it like this, it works when data and mask are all MetaTensor.
 import torch from monai.data import MetaTensor data = MetaTensor(torch.zeros([64,64,3])) mask = MetaTensor(torch.zeros([64,64,3])) mask[:, 0, :] = 2 data[mask &gt;1 ] =1 print(type(data), type(mask)) output --&gt; <class 'monai.data.meta_tensor.MetaTensor'&gt; <class 'monai.data.meta_tensor.MetaTensor'&gt;  
Hope it can help you, thanks!
